### PR TITLE
Handle zero-prefixed usernames in UserProfile

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/6007

I checked all other usage of `parseInt()` and I don't think we need to fix anything else. It's surprising that `parseInt('0something', 10)` returns `0` 🤡 👯

### CODE REVIEW

File: src/amo/components/UserProfile/index.js [hunk 1]
```diff
@@ -448,7 +448,7 @@ export function mapStateToProps(state: AppState, ownProps: Props) {
   // `getUserByUsername()` requires a string as second argument.
   let user = getUserByUsername(state.users, `${username}`);
 
-  if (!user) {
+  if (!user && /^[0-9]+$/.test(username)) {
```
Comment (willdurand) [inline src/amo/components/UserProfile/index.js]: I have been thinking about this patch and one edge case would be `username = 0`.

Comment (kumar303) [reply src/amo/components/UserProfile/index.js]: Any username consisting of all numbers would defeat it, yes. I tried it and it's not possible:

screenshot 2018-08-23 12 33 44

But I tried many numbers at first (including `0`) and they were all taken usernames 😛 That might just be dirty dev data, I'm not sure.

Comment (willdurand) [reply src/amo/components/UserProfile/index.js]: Yep but 0 is a special case, no? It will throw an error because of the invariant.

Comment (kumar303) [reply src/amo/components/UserProfile/index.js]: Ah, right, probably. We'll see if `0` sends us a bug report. Until then I'm not worried.
